### ISSUE TITLE
`RobotsTxtMiddleware` sometimes misbehaves

### ISSUE THREAD

**Issue by starrify** [scrapy/scrapy#2014]
Some requests might have been fetched before `RobotsTxtMiddleware` could start filtering them:

```
...
43:	2016-05-27 05:13:25	DEBUG	[scrapy.core.engine] Crawled (200) <GET http://i1.adis.ws/s/boohooamplience/szz99792_ms.json?protocol=http&deep=true> (referer: http://www.boohoo.com/new-in/yolanda-boutique-bandage-neon-cut-out-swimsuit/invt/szz99792)
44:	2016-05-27 05:13:27	DEBUG	[scrapy.core.engine] Crawled (200) <GET http://mark.reevoo.com/robots.txt> (referer: None)
45:	2016-05-27 05:13:27	DEBUG	[scrapy.core.engine] Crawled (200) <GET http://mark.reevoo.com/reevoomark/product_summary?locale=en-GB&sku=szz99940&trkref=BHO&callback=> (referer: http://i1.adis.ws/s/boohooamplience/szz99940_ms.json?protocol=http&deep=true)
46:	2016-05-27 05:13:27	DEBUG	[scrapy.downloadermiddlewares.robotstxt] Forbidden by robots.txt: <GET http://mark.reevoo.com/reevoomark/product_summary?locale=en-GB&sku=szz99792&trkref=BHO&callback=>
47:	2016-05-27 05:13:27	DEBUG	[scrapy.core.scraper] Scraped from <200 http://mark.reevoo.com/reevoomark/product_summary?locale=en-GB&sku=szz99940&trkref=BHO&callback=> 
...
```

And once that happens, following `IgnoreRequest` exceptions are not handled (logs from the same job):

```
...
62:	2016-05-26 20:11:55	ERROR	[scrapy.core.scraper] Spider error processing <GET http://mark.reevoo.com/reevoomark/product_summary?locale=en-GB&sku=mzz83566&trkref=BHO&callback=> (referer: http://i1.adis.ws/s/boohooamplience/mzz83566_ms.json?protocol=http&deep=true)
IgnoreRequest
...
66:	2016-05-26 20:11:57	ERROR	[scrapy.core.scraper] Spider error processing <GET http://mark.reevoo.com/reevoomark/product_summary?locale=en-GB&sku=dzz80021&trkref=BHO&callback=> (referer: http://i1.adis.ws/s/boohooamplience/dzz80021_ms.json?protocol=http&deep=true)
IgnoreRequest
...
70:	2016-05-26 20:12:00	ERROR	[scrapy.core.scraper] Spider error processing <GET http://mark.reevoo.com/reevoomark/product_summary?locale=en-GB&sku=dzz78696&trkref=BHO&callback=> (referer: http://i1.adis.ws/s/boohooamplience/dzz78696_ms.json?protocol=http&deep=true)
IgnoreRequest
...
```

For details of the job, please check https://staging.scrapinghub.com/p/53580/job/17/5/#/log (sorry but the link is not publicly accessible)

**Comment by kmike** [comment]
Hey @starrify, 

I think the reason is that you're using Scrapy 1.0.x; it was a known limitation of robots.txt middleware - see Scrapy 1.0 [docs](http://doc.scrapy.org/en/1.0/topics/downloader-middleware.html#module-scrapy.downloadermiddlewares.robotstxt). The issue is fixed in Scrapy 1.1 by https://github.com/scrapy/scrapy/pull/1473, so there is no warning in 1.1 [docs](http://doc.scrapy.org/en/1.1/topics/downloader-middleware.html#module-scrapy.downloadermiddlewares.robotstxt).

**Comment by starrify** [comment]
:+1: Thanks @kmike! :D

**Comment by starrify** [comment]
> The issue is fixed in Scrapy 1.1 by #1473
No wonder why I checked the code and couldn't find what's going wrong..
